src/zarr/metadata/v3.py

```diff
@@ -636,3 +314,83 @@ def parse_fill_value_v3(
         msg = f"Cannot parse non-string sequence {fill_value} as a scalar with type {dtype}."
         raise TypeError(msg)
     return dtype.type(fill_value)
+
+
+def parse_dimension_names(data: None | Iterable[str | None]) -> tuple[str | None, ...] | None:
+    if data is None:
+        return data
+    elif all(isinstance(x, type(None) | str) for x in data):
+        return tuple(data)
+    else:
+        msg = f"Expected either None or a iterable of str, got {type(data)}"
+        raise TypeError(msg)
+
+
+class DataType(Enum):
+    bool = "bool"
+    int8 = "int8"
+    int16 = "int16"
+    int32 = "int32"
+    int64 = "int64"
+    uint8 = "uint8"
+    uint16 = "uint16"
+    uint32 = "uint32"
+    uint64 = "uint64"
+    float32 = "float32"
+    float64 = "float64"
+
+    @property
+    def byte_count(self) -> int:
+        data_type_byte_counts = {
```

Comment:
I know this wasn't introduced here but we should pull all of these dicts into module level variables. No need to recreate them every time we use the datatype.